lib/pure/os.nim

```diff
@@ -930,15 +929,37 @@ proc getConfigDir*(): string {.rtl, extern: "nos$1",
     result = getEnv("XDG_CONFIG_HOME", getEnv("HOME").string / ".config").string
   result.normalizePathEnd(trailingSep = true)
 
+from std/private/winutils import nil
+
+when defined(windows):
+  proc getTempPath(
+    nBufferLength: winutils.DWORD, lpBuffer: WideCString
+  ): winutils.DWORD {.stdcall, dynlib: "kernel32.dll", importc: "GetTempPathW".} =
+    ## Retrieves the path of the directory designated for temporary files.
+
+template getEnvImpl(result: var string, tempDirList: openArray[string]) =
+  for dir in tempDirList:
+    if existsEnv(dir):
+      result = getEnv(dir)
+      break
+
+template getTempDirImpl(result: var string) =
+  when defined(windows):
+    getEnvImpl(result, ["TMP", "TEMP", "USERPROFILE"])
+  else:
+    getEnvImpl(result, ["TMPDIR", "TEMP", "TMP", "TEMPDIR"])
```

Comment:
for posix, what's your justification for `"TEMP", "TMP", "TEMPDIR"` ?

https://unix.stackexchange.com/questions/352107/generic-way-to-get-temp-path only mentions TMPDIR

Reply:
https://github.com/libuv/libuv/blob/39968db6434f16f9fbd3c78f6c7392fd7ae9074e/src/unix/core.c#L1109